root/static/scripts/release/components/ReleaseRelationshipEditor.js

```diff
@@ -1552,6 +1563,29 @@ let ReleaseRelationshipEditor: React.AbstractComponent<{}, void> = (
     };
   }, [dispatch, state]);
 
+  const hasPendingEdits = React.useMemo(() => {
+    return stateHasPendingEdits(state);
+  }, [state]);
+
+  React.useEffect(() => {
+    const beforeUnload = function (event: BeforeUnloadEvent) {
+      if (state.submissionInProgress) {
+        return undefined;
+      }
+      if (hasPendingEdits) {
+        // Modern browsers don't actually display this string for security.
+        event.returnValue =
+          'All of your changes will be lost if you leave this page.';
```

Comment:
This is the string that the old code used, but I removed the `l()` call from it, because as noted by the comment, browsers don't actually display it anymore.

Reply:
You have to return a string for it to work, so you could just return `''`, but I kept the old one since it's self-documenting (and maybe some non-mainstream browsers still show it -- it wasn't that long ago that the mainstream ones did).